web/src/engine/websites/LineWebtoon.ts

```diff
@@ -0,0 +1,261 @@
+        super(id, label, url, ...tags);
+    }
+    public override get Icon() {
+        return icon;
+    }
+
+    public override ValidateMangaURL(url: string): boolean {
+        return this.mangaRegexp.test(url);
+    }
+
+    public override async FetchManga(provider: MangaPlugin, url: string): Promise<Manga> {
+        const manga = await Common.FetchMangaCSS.call(this, provider, url, this.queryMangaTitleURI, this.mangaLabelExtractor, true, false);
+        try {
+            const languageCode = url.match(this.languageRegexp)[1];
+            manga.Tags.Value.concat([mangasLanguageMap[languageCode]]);
+        } catch { }
+        return manga;
+    }
+
+    public override async FetchChapters(manga: Manga): Promise<Chapter[]> {
+        const chapterList: Chapter[] = [];
+        for (let page = 1, run = true; run; page++) {
+            const chapters = await this.GetChaptersFromPage(manga, page);
+            chapterList.isMissingLastItemFrom(chapters) ? chapterList.push(...chapters) : run = false;
+        }
+        return chapterList.distinct();
+    }
+
+    private async GetChaptersFromPage(manga: Manga, page: number): Promise<Chapter[]> {
+        const data = await FetchCSS<HTMLAnchorElement>(new Request(new URL(`${manga.Identifier}&page=${page}`, this.URI)), this.queryChapters);
+        return data.map(element => {
+            const { id, title } = ChapterExtractor.call(this, element);
+            return new Chapter(this, manga, id, title);
+        });
+    }
+
+    public override async FetchPages(chapter: Chapter): Promise<Page[]> {
+        const data = await FetchWindowScript(new Request(new URL(chapter.Identifier, this.URI)), this.pageScript, 1500);
+        if (!Array.isArray(data)) return [];
+        return typeof data[0] === 'string' ? (data as Array<string>).map(page => new Page(this, chapter, new URL(page))) : this.CreatePagesfromData(chapter, data as PageData[]);
+    }
+
+    private CreatePagesfromData(chapter: Chapter, data: PageData[]): Page<PageData>[] {
+        return data.map(page => new Page<PageData>(this, chapter, new URL(this.URI), { Referer: this.URI.origin, ...page }));
+    }
+
+    public override async FetchImage(page: Page<PageData>, priority: Priority, signal: AbortSignal): Promise<Blob> {
+        if (!page.Parameters?.width) return await Common.FetchImageAjax.call(this, page, priority, signal, true);
+        const payload = page.Parameters;
+        return this.imageTaskPool.Add(async () => {
+            return DeScramble(new ImageData(payload.width, payload.height,), async (_, ctx) => {
+                ctx.canvas.width = payload.width;
+                ctx.canvas.height = payload.height;
+
+                if (payload.background.image) {
+                    const image = await LoadImage(payload.background.image);
+                    ctx.canvas.width = image.width;
```

Comment:
This will replace the previously assigned width/height, is this by intention?

Reply:
It was copied straight from Hakuneko and it would make senses that a background image would be the actual image size, i just wont re-invent the wheel.